### PR TITLE
Update for upcoming variable changes

### PR DESCRIPTION
**Pull request by gschier**
This PR makes the necessary updates to document the upcoming changes to how plugin variables are referenced.

- Reworked Variables doc and screenshots
- Updated plugin doc pages
- A few other minor changes

### CODE REVIEW

File: src/docs/develop/variables.md [hunk 1]
```diff
@@ -2,46 +2,52 @@
...
+You can view all variables for the current environment using the [Railway CLI](/develop/cli) with 
+`railway variables` and change the environment with `railway environment`.
 
-You can click the `RAW Editor` link to open up a modal where you can paste in a list of variables separated by new line. This also allows you to bulk edit your variables in a familiar way like you would an `.env` file or even edit them via JSON notation.
+### Raw Editor 
 
-## Automatic Detection of Environment Variables
+If you already have a `.env` file, or simply prefer to edit text, the Raw Editor can be used to edit variables in either `.env` or `.json` format.
+
+**_NOTE: If you initialize a new project from the CLI and an .env file is detected, Railway will prompt you to import it into your new project._**
```
Comment (ndneighbor) [inline src/docs/develop/variables.md:30]: This functionality is no longer in the new CLI. We need to re-add this.

**Review by ndneighbor — APPROVED** [review]
LGTM

Not going to lie, I was a bit confused on the copy on the connection part but it describes the behavior well.

File: src/docs/databases/mongodb.md [hunk 2]
```diff
@@ -7,13 +7,21 @@ MongoDB database with zero configuration.
...
 
-- MONGOHOST
-- MONGOPORT
-- MONGOUSER
-- MONGOPASSWORD
-- MONGO_URL
+- Add a [Reference Variable](/develop/variables#reference-variables) to a service
+- Run `railway connect` to start a `mongodb` shell
+
+The following variables are made available to reference in your services:
```
Comment (ndneighbor) [inline src/docs/databases/mongodb.md:15]: We should disambiguate the ways on how people connect to the DB. We should make it explicit that exposing the variable to the service will give the service access to the DB.

File: src/docs/develop/variables.md (hunk 1 above)
Comment (gschier) [inline src/docs/develop/variables.md:30]:
```suggestion
```

**Pull request merged**